src/AdminAan/SFA.DAS.AdminAan/Infrastructure/IAanHubRestApiClient.cs

```diff
@@ -30,7 +30,7 @@ public interface IAanHubRestApiClient
 
     [Get("calendarEvents/{calendarEventId}")]
     [AllowAnyStatusCode]
-    Task<GetCalendarEventQueryResult> GetCalendarEvent([Header(Constants.ApiHeaders.RequestedByMemberIdHeader)] Guid requestedByMemberId, [Path] Guid calendarEventId, CancellationToken cancellationToken);
+    Task<GetCalendarEventQueryResult?> GetCalendarEvent([Header(Constants.ApiHeaders.RequestedByMemberIdHeader)] Guid requestedByMemberId, [Path] Guid calendarEventId, CancellationToken cancellationToken);
```

Comment:
The intention is not clear here. With AllowAnyStatusCode you have to accept a Response object which is missing. In the handler there is no check for status code ok, but the ApiClient has nullable return type. 
If you are sure that a calendar event will always be there, remove AllowAnyStatusCode and make return type not nullable, that way your handler can always expect a response otherwise the client will raise exception.